include/boost/url/impl/url.ipp

```diff
@@ -145,194 +145,10 @@ relative(
-
-    // Function to advance the dotdot segments
-    decode_view dotdot("..");
-    decode_view dot(".");
-    auto consume_dots = [dotdot, dot](
-        segments_view::iterator& first,
-        segments_view::iterator last)
-    {
-        if (*first == dotdot ||
-            *first == dot)
-        {
-            ++first;
-            return true;
-        }
-        auto it = std::next(first);
-        std::size_t l = 1;
-        while (it != last)
-        {
-            if (*it == dotdot)
-            {
-                if (--l == 0)
-                {
-                    ++it;
-                    first = it;
-                    break;
-                }
-            }
-            else if (*it != dot)
-            {
-                ++l;
-            }
-            ++it;
-        }
-        return first == it;
-    };
-
-    // 1. Find the longest common path
-    while (
-        it0 != last0 &&
-        it1 != last1)
-    {
-        if (consume_dots(it0, last0))
-            continue;
-        if (consume_dots(it1, last1))
-            continue;
-        if (*it0 == *it1)
-        {
-            ++it0;
-            ++it1;
-        }
-        else
-        {
-            break;
-        }
-    }
-
-    // 1.b Check if parent paths are the same
-    if (it0 == last0 &&
-        it1 == last1 &&
-        it0 != end0 &&
-        it1 != end1 &&
-        *it0 == *it1)
-    {
-        // Return empty path
-        if (href.has_query())
-            dest.set_encoded_query(href.encoded_query());
-        else
-            dest.remove_query();
-        if (href.has_fragment())
-            dest.set_encoded_fragment(href.encoded_fragment());
-        else
-            dest.remove_fragment();
-        return {};
-    }
-
-    // 2. Append ".." for each segment left in base
-    segments_encoded segs = dest.encoded_segments();
-    if (it0 != end0)
-    {
-        dest.set_path_absolute(false);
-        while (it0 != last0)
-        {
-            if (*it0 == dotdot)
-            {
-                if (!segs.empty())
-                    segs.pop_back();
-            }
-            else if (*it0 != dot)
-            {
-                segs.push_back(dotdot.encoded());
-            }
-            ++it0;
-        }
-    }
-
-    // 3. Append segments left from the reference
-    while (it1 != end1)
-    {
-        if (*it1 == dotdot)
-        {
-            if (!segs.empty())
-                segs.pop_back();
-        }
-        else if (*it1 != dot)
-        {
-            segs.push_back((*it1).encoded());
-        }
-        ++it1;
-    }
-
-    // Query and fragment comes from reference
-    if (href.has_query())
-        dest.set_encoded_query(href.encoded_query());
-    else
-        dest.remove_query();
-
-    if (href.has_fragment())
-        dest.set_encoded_fragment(href.encoded_fragment());
-    else
-        dest.remove_fragment();
-
-    return {};
+    dest.copy(base);
+    return dest.relative(href);
```

Comment:
`relative` is the name? That is kind of obscure... doesn't really suggest what it does. Why are we adding features

Reply:
> relative is the name? That is kind of obscure... doesn't really suggest what it does. 

URL.js uses `relative_to`. I can't think of anything better than `relative`/`relative_to`

> Why are we adding features

I don't know. I was assigned this issue and started writing the code at some point.